count.go

```diff
@@ -0,0 +1,51 @@
+package storm
+
+import (
+	"fmt"
+
+	"github.com/boltdb/bolt"
+	"github.com/fatih/structs"
+)
+
+// Count counts all the records of a bucket
+func (n *Node) Count(data interface{}) (int, error) {
+	if !structs.IsStruct(data) {
+		return 0, ErrBadType
+	}
+
+	info, err := extract(data)
+	if err != nil {
+		return 0, err
+	}
+
+	var count int
+	if n.tx != nil {
+		err = n.count(n.tx, info, &count)
+		return count, err
+	}
+
+	err = n.s.Bolt.View(func(tx *bolt.Tx) error {
+		return n.count(tx, info, &count)
+	})
+	return count, err
+}
+
+func (n *Node) count(tx *bolt.Tx, info *modelInfo, count *int) error {
+	bucket := n.GetBucket(tx, info.Name)
+	if bucket == nil {
+		return fmt.Errorf("bucket %s not found", info.Name)
+	}
+
+	*count = 0
+	c := bucket.Cursor()
+	for k, v := c.First(); k != nil && v != nil; k, v = c.Next() {
+		(*count)++
```

Comment:
The thing with your implementation is that you also count sub buckets and indexes.
Your tests works because the indexes are prefixed with `__` so they are put at the end of the bucket, but if i change the user name to "___john" it breaks.

In every bucket that uses indexes, Storm creates one sub bucket per index. So if you have 100 regular items in your bucket, in reality you have 100 + the number of indexes.
(example for User: it has one index for `Name`, one for `DateOfBirth` and one for `Slug` => 103 items)

When using a `Cursor`, if `Next` finds a sub bucket it returns a `nil` value so you can easily skip sub buckets by adding a simple `if` statement in you loop and changing your condition:

```go
for k, v := c.First(); k != nil; k, v = c.Next() {
 if v == nil {
   continue
 }
 (*count)++
}
```